marimo/_convert/non_marimo_python_script.py

```diff
@@ -20,22 +79,31 @@ def convert_non_marimo_python_script_to_notebook_ir(
     2. Otherwise, puts all content into a single cell
     3. Preserves PEP 723 inline script metadata if present
     """
+
     if "# %%" in source:
-        try:
-            import jupytext  # type: ignore[import-untyped]
-        except ImportError as e:
-            raise ImportError(
-                "Converting py:percent format requires jupytext"
-            ) from e
-        notebook = jupytext.reads(source, fmt="py:percent")
-    else:
-        notebook = {"cells": [{"source": source, "cell_type": "code"}]}
+        return convert_pypercent_script_to_notebook_ir(source)
+    return convert_python_block_to_notebook_ir(source)
 
-    notebook_str = json.dumps(notebook)
 
-    ir = convert_from_ipynb_to_notebook_ir(notebook_str)
-    _transform_main_blocks(ir)
-    return ir
+def convert_non_marimo_script_to_notebook_ir(
+    source: str,
+) -> NotebookSerialization:
+    """Convert a non-marimo script to marimo notebook IR.
+
+    This is a convenience function that turns any string into a
+    marimo notebook.
+    """
+    try:
+        return convert_non_marimo_python_script_to_notebook_ir(source)
+    except ImportError:
+        pass
+    try:
+        import ast
+
+        ast.parse(source)  # Validate if it's valid Python code
+        return convert_python_block_to_notebook_ir(source)
+    except SyntaxError:
+        return convert_script_block_to_notebook_ir(source)
```

Comment:
instead of all these try/except blocks and check, could we instead create a common convert:

```
class NotebookConverter:
     def try_convert(): IR | None

converters = [OneCoverter, TwoCoverter]
for converter in converts:
   if result := convert.try_convert(source)
           return result
```

also could use some tests for these new functions

Reply:
Added the tests. Agreed that pattern is cleaner, but think it adds a lot more boiler plate for this one off code path. We can revisit if conversion become even hairier